tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_report.py

```diff
@@ -0,0 +1,81 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+from marionette_driver import expected, By, Wait
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.base import Base
+from gaiatest.apps.messages.app import Messages
+
+
+class TestSmsReport(GaiaTestCase):
+
+    def test_sms_report(self):
+        """This test sends a text message to itself. It waits for a reply message.
+           Then it checks the info in the report.
+        """
+        _text_type = (By.CSS_SELECTOR, 'p[data-l10n-id="message-type-sms"]')
+        _sender = (By.CSS_SELECTOR, 'a[data-name]')
+        _receiver = (By.CSS_SELECTOR, 'a.suggestion.js-contact-info')
+        
+        # Configuring the test
+        time_sent = time.time()
+        _text_message_content = "Automated Test %s" % str(time_sent)
+        _test_phone_number = self.testvars['phone_numbers']
+        
+        error_range_sent_in_seconds = 1200
+        error_range_rec_in_seconds = 3000
+        
+        # Sending the message
+        self.messages = Messages(self.marionette)
+        self.messages.launch()
+
+        new_message = self.messages.create_new_message(recipients=_test_phone_number,
+                                                       message=_text_message_content)
+        self.message_thread = new_message.tap_send()
+        self.message_thread.wait_for_received_messages()
+
+        last_received_message = self.message_thread.received_messages[-1]
+        last_message = self.message_thread.all_messages[-1]
+        
+        # Long Tap on received message and check the report: SMS sent received and time
+        report = last_received_message.open_report()
+
+        # Checking that the data in the received report is correct
+        # Type of message
+        self.assertTrue(report.is_message_an_sms)
+
+        # From, sender is the right one
+        self.assertEqual(_test_phone_number, report.sender)
+
+        # Sent date
+        time_sent_err = (time_sent + error_range_sent_in_seconds) 
+        time_sent_err_str = str(time_sent_err) 
+        self.assertLessEqual(str(report.is_date_sent_ok),time_sent_err_str)
```

Comment:
Having to convert a time in string is odd. The python class supports the <, <=, >, >= operators. I guess your problem is that `report.is_date_sent_ok` return something like 1445952146000, whereas the time object returns 1445953315.97.

One way to solve this is issue, is to make `report.is_date_sent_ok` return a time object. You can do it this way:
```py
@property
def is_date_sent_ok(self):  # name to be changed
    timestamp_in_milliseconds = self.marionette.find_element(*self._time_sent_locator).get_attribute('data-l10n-date')
    return time.ctime(timestamp_in_milliseconds // 1000)
```
By the way, you'll have twice the same formula to apply. In this case, I'd suggest:
```py
@property
def is_date_sent_ok(self):  # name to be changed
    return _time_from_locator(self._time_sent_locator)
# same thing for the receiver

def _time_from_locator(self, locator):
   timestamp_in_milliseconds = self.marionette.find_element(*locator).get_attribute('data-l10n-date')
    return time.ctime(timestamp_in_milliseconds // 1000)
```